test/soak/serve_hostnames/serve_hostnames.go

```diff
@@ -120,27 +141,36 @@ func main() {
 		for j := 0; j < *perNode; j++ {
 			podName := fmt.Sprintf("serve-hostname-%d-%d", i, j)
 			podNames = append(podNames, podName)
-			glog.Infof("Creating pod %s/%s on node %s", ns, podName, node.Name)
-			_, err := c.Pods(ns).Create(&api.Pod{
-				ObjectMeta: api.ObjectMeta{
-					Name: podName,
-					Labels: map[string]string{
-						"name": "serve-hostname",
+			// Make several attempts
+			for start := time.Now(); time.Since(start) < podCreateTimeout; time.Sleep(2 * time.Second) {
+				glog.Infof("Creating pod %s/%s on node %s", ns, podName, node.Name)
+				t := time.Now()
+				_, err = c.Pods(ns).Create(&api.Pod{
+					ObjectMeta: api.ObjectMeta{
+						Name: podName,
+						Labels: map[string]string{
+							"name": "serve-hostname",
+						},
 					},
-				},
-				Spec: api.PodSpec{
-					Containers: []api.Container{
-						{
-							Name:  "serve-hostname",
-							Image: "kubernetes/serve_hostname:1.1",
-							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+					Spec: api.PodSpec{
+						Containers: []api.Container{
+							{
+								Name:  "serve-hostname",
+								Image: "kubernetes/serve_hostname:1.1",
+								Ports: []api.ContainerPort{{ContainerPort: 9376}},
+							},
 						},
+						Host: node.Name,
 					},
-					Host: node.Name,
-				},
-			})
+				})
+				glog.Infof("Pod create %s/%s request took %v", ns, podName, time.Since(t))
```

Comment:
Again, I prefer it where it is. I want to time the requests (regardless of success or failure).